honeybee/boundarycondition.py

```diff
@@ -1,116 +1,174 @@
     def wind_exposure(self):
+        """Get or set a boolean noting whether the boundary is exposed to wind."""
         return self._wind_exposure
 
     @wind_exposure.setter
     def wind_exposure(self, value):
         assert isinstance(value, bool), \
-            'Input value must be a Boolean not {}.'.format(type(value))
+            'Input wind_exposure must be a Boolean. Got {}.'.format(type(value))
         self._wind_exposure = value
 
-    @property
-    def boundary_condition_object(self):
-        return self._boundary_condition_object
-
-    @boundary_condition_object.setter
-    def boundary_condition_object(self, bco):
-        self._boundary_condition_object = bco
-
     @property
     def view_factor(self):
+        """Get or set the view factor to the ground as a number or 'autocalculate'."""
         return self._view_factor
 
     @view_factor.setter
     def view_factor(self, vf):
         if vf == 'autocalculate':
             self._view_factor = vf
         else:
-            self._view_factor = float(vf)
+            self._view_factor = float_in_range(vf, 0.0, 1.0, 'view factor to ground')
 
     @property
     def sun_exposure_idf(self):
+        """Text string for sun exposure, which is write-able into an IDF."""
         return 'NoSun' if not self.wind_exposure else 'SunExposed'
 
     @property
     def wind_exposure_idf(self):
+        """Text string for wind exposure, which is write-able into an IDF."""
         return 'NoWind' if not self.wind_exposure else 'WindExposed'
 
-    @property
-    def boundary_condition_object_idf(self):
-        return '' if not self.boundary_condition_object else \
-            self.boundary_condition_object.name
-
     def to_dict(self, full=False):
         """Boundary condition as a dictionary.
-        
-        args:
-            full: Set to True to get the full dictionary which includes energy simulation
-            specific keys such as sun_exposure, wind_exposure and view_factor. Default is
-            set to False.
-        
-        returns:
-            A dictionary.
+
+        Args:
+            full: Set to True to get the full dictionary which includes energy
+                simulation specific keys such as sun_exposure, wind_exposure and
+                view_factor. Default: False.
         """
         if full:
-            return {
-                'name': self.name,
-                'bc_object': self.boundary_condition_object_idf,
+            bc_dict = {
+                'type': self.name,
                 'sun_exposure': self.sun_exposure_idf,
                 'wind_exposure': self.wind_exposure_idf,
                 'view_factor': self.view_factor
             }
         else:
-            return {
-                'name': self.name,
-                'bc_object': self.boundary_condition_object_idf
+            bc_dict = {
+                'type': self.name,
             }
+        try:
+            bc_dict['bc_object'] = self.boundary_condition_object
```

Comment:
```suggestion
            bc_dict['bc_object'] = self.boundary_condition_object.to_dict(full)
```

Reply:
This is not correct since, after one of our conversations, we agreed that the boundary condition objects are just going to be the names of the objects and not entire objects.  Seeing this example actually makes me think that we made the right decision.  If you think about it, you can see that there would be infinite recursion had the boundary condition object actually been an object that we were sending `to_dict` like what you have here. Having the adjacent face going `to_dict` would cause that adjacent Face's boundary condition to also be called `to_dict`, which would call the original Face's boundary condition `to_dict` and so on.

Reply:
You are right. Let's see how this implementation will lead us. I know the logic that lead us to this decision but it still doesn't _feel_ right.